### PR TITLE
net: wifi: Add support for WPA supplicant

### PR DESCRIPTION
**Pull request by krish2718**
If WPA supplicant is enabled at build time, then use the WPA supplicant APIs, else use offload driver APIs.

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -68,12 +68,14 @@ static void handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
 
 	if (scan_result == 1U) {
 		print(context.sh, SHELL_NORMAL,
-		      "\n%-4s | %-32s %-5s | %-4s | %-4s | %-5s    | %s\n",
-		      "Num", "SSID", "(len)", "Chan", "RSSI", "Sec", "MAC");
+		      "\n%-4s | %-32s %-5s | %-13s | %-4s | %-15s | %s\n",
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:71]: The changes in wifi_shell.c will come with an upmerge, let's not introduce them here. And for the future - you should rather cherry-pick upstream commits and tag them with `[nrf fromtree]` prefix instead of creating your own commits.

File: subsys/net/l2/wifi/wifi_mgmt.c [hunk 2]
```diff
@@ -12,19 +12,25 @@ LOG_MODULE_REGISTER(net_wifi_mgmt, CONFIG_NET_L2_WIFI_MGMT_LOG_LEVEL);
...
 			void *data, size_t len)
 {
 	struct wifi_connect_req_params *params =
 		(struct wifi_connect_req_params *)data;
 	const struct device *dev = net_if_get_device(iface);
+#ifndef CONFIG_WPA_SUPP
 	struct net_wifi_mgmt_offload *off_api =
 		(struct net_wifi_mgmt_offload *) dev->api;
 
+
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_mgmt.c:29]: Please keep the diff minimal, no unnecessary changes.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c]: Sure, done. I have dropped these changes as they are being taken care in upmerge.

**Review by sachinthegreen — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
